packages/fetchai/connections/http/connection.py

```diff
@@ -0,0 +1,384 @@
+                    # handler = SimpleHTTPServer.SimpleHTTPRequestHandler
+                    # handler = HTTPHandler.set_channel(self)
+                    # handler.set_channel(self)
+                    self.httpd = HTTPServer(
+                        (self.host, self.port), HTTPHandlerFactory(self)
+                    )
+                    # self.httpd = HTTPServer((self.host, self.port), handler)
+                    logger.info(f"HTTP Server has connected to port {self.port}.")
+                    self.httpd.serve_forever()
+                except OSError:
+                    logger.error(
+                        f"{self.host}:{self.port} is already in use, please try another Socket."
+                    )
+
+    def disconnect(self) -> None:
+        """
+        Disconnect.
+
+        Shut-off the HTTP Server and then HTTP Channel.
+        """
+        with self.lock:
+            if not self.stopped:
+                self.httpd.shutdown()
+                logger.info(f"HTTP Server has shutdown on port {self.port}.")
+                self.stopped = True
+                # self.thread.join()
+
+
+def HTTPHandlerFactory(channel: HTTPChannel):
+    class HTTPHandler(BaseHTTPRequestHandler):
+        def __init__(self, *args, **kwargs):
+            self._channel = channel
+            super(HTTPHandler, self).__init__(*args, **kwargs)
+
+        def do_HEAD(self):
+            self.send_response(200)
+            self.send_header("Content-type", "text/html")
+            self.end_headers()
+
+        def do_GET(self):
+            """Respond to a GET request."""
+            self.send_response(200)
+            self.send_header("Content-type", "text/html")
+            self.end_headers()
+
+            parsed_path = urlparse(self.path)
+            url = parsed_path.path
+            # url = parsed_path.geturl()
+            param = parse_qs(parsed_path.query)
+            param = json.dumps(param)
+
+            self._channel._process_request("GET", url, param)
+
+        def do_POST(self):
+            """Respond to a POST request."""
+            self.send_response(200)
+            self.send_header("Content-type", "text/html")
+            self.end_headers()
+
+            parsed_path = urlparse(self.path)
+            url = parsed_path.path
+            # url = parsed_path.geturl()
+            param = parse_qs(parsed_path.query)
+            param = json.dumps(param)
+
+            content_length = int(self.headers["Content-Length"])
+            body = self.rfile.read(content_length).decode()
+
+            self._channel._process_request("POST", url, param, body)
+
+    return HTTPHandler
+
+
+class HTTPConnection(Connection):
+    """Proxy to the functionality of the web RESTful API."""
+
+    def __init__(
+        self,
+        address: Address,
+        host: str = "",
+        port: int = 10000,
+        api_path: str = None,  # Directory path of the API YAML file.
```

Comment:
I think you'd want it to be `api_specification_path: Optional[str]`

Reply:
I changed it to `api_spec_path: Optional[str]`, will be in next commit.

Reply:
For HTTPConnection, I changed it back so that it gives the user flexibility not to pass in anything into `api_spec_path` (although that may not be a good idea in terms of verification: https://stackoverflow.com/questions/51848288/in-python-should-we-set-optionalstr-none-in-the-argument-list 

But still good for demo purposes to showcase the differences between passing in the YAML file versus not.